Templates/test_template.html

```diff
@@ -0,0 +1,74 @@
+<html>
+    <head>
+        <title>&#x7b;title}={title}</title>
+        <style>
+            body {
+                font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
+            }
+            div {
+                border: solid 1px #ccf;
+                margin: 2px;
+                padding: 3px;
+            }
+        </style>
+    </head>
+    <body>
+<!--
+  The four types of pages used in conditional properties are:
+    home     = the single master home page
+               rendered as "$TargetDir/index.html"
+    contents = the single master index/contents page
+               rendered as "$TargetDir/contents.html"
+    module   = the module index pages, one per module
+               rendered as "$TargetDir/$nsName/$modName/index.html"
+    function = the function pages, one per exported function
+               rendered as "$TargetDir/$nsName/$modName/$funName.html"
+  -->
+
+        <div>
+            <code>Conditional Properties</code> by Page Type:
+            {ifdef home}     <b><code> &#x7b;ifdef home}     </code></b> {endif home}
+            {ifdef contents} <b><code> &#x7b;ifdef contents} </code></b> {endif contents}
+            {ifdef module}   <b><code> &#x7b;ifdef module}   </code></b> {endif module}
+            {ifdef function} <b><code> &#x7b;ifdef function} </code></b> {endif function}
+        </div>
+
+        {ifdef module}<!--{endif module}
+        {ifdef function}<!--{endif function}
```

Comment:
This works because it's before the render of the HTML output, that is, it's pre-processed by the POSH code with string substitutions, so afterwards, only one of these (or neither of them) will ever be output as legal HTML.  The POSH pre-processing just treats the template bodies as large strings.